blueocean-rest/pom.xml

```diff
@@ -24,4 +24,47 @@
           <artifactId>capability-annotation</artifactId>
       </dependency>
   </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <dependencies>
+                    <dependency>
+                        <groupId>de.skuzzle.enforcer</groupId>
+                        <artifactId>restrict-imports-enforcer-rule</artifactId>
+                        <version>0.7.0</version>
+                    </dependency>
+                </dependencies>
+                <executions>
+                    <execution>
+                        <id>check-no-jenkins-imports</id>
+                        <phase>process-sources</phase>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <restrictImports implementation="de.skuzzle.enforcer.restrictimports.RestrictImports">
+                                    <basePackage>**</basePackage>
```

Comment:
I guess we can do that, Item is one of the thing most blue extensions depend on, although it pulls in lots of other dependencies but not too bad. Where as Run (needed by BlueRunFactory) has too many dependencies and its like opening up to broader set of dependencies in Jenkins core.